.github/workflows/run-tests.yml

```
@@ -77,6 +77,8 @@ jobs:
         xvfb_pid=$!
 
         cd testing
-        ./run3 run_tests.py
+        ./run3 run_tests.py || touch FAILED
 
         kill -s SIGTERM "${xvfb_pid}"
+        [ -f FAILED ] && exit 1
+        exit 0
```

I believe this is bash, so can probably use `trap` (and also change `set -x` to `set -ex` to error directly)?
```suggestion
        xvfb_pid=$!
        trap "kill -s SIGTERM ${xvfb_pid}" EXIT

        cd testing
        ./run3 run_tests.py
```

---

I mean, sure, is it *better* though? I kinda like simple. :P